### PR TITLE
makes L.Browser mutable

### PR DESCRIPTION
**Pull request by bozdoz**
As suggested in https://github.com/Leaflet/Leaflet/pull/7328

tests can't test for detectRetina without mutable L.Browser.

### CODE REVIEW

**Review by johnd0e — DISMISSED** [review]

File: src/layer/tile/TileLayer.js [hunk 1]
```diff
@@ -93,10 +93,10 @@ export var TileLayer = GridLayer.extend({
 
 			if (!options.zoomReverse) {
 				options.zoomOffset++;
-				options.maxZoom--;
+				options.maxZoom = Math.max(options.minZoom, options.maxZoom - 1);
```
Comment (bozdoz) [inline src/layer/tile/TileLayer.js:96]: shoot; I accidentally included the fix to #7328. I will revert this @johnd0e

Comment (bozdoz) [reply src/layer/tile/TileLayer.js:96]: For some reason my last commit isn't being added to this PR; perhaps because you've approved it @johnd0e

Comment (johnd0e) [reply src/layer/tile/TileLayer.js:96]: > perhaps because you've approved it @johnd0e

I don't think that there can be any connection. But whatever..

Comment (bozdoz) [reply src/layer/tile/TileLayer.js]: Looks like it worked! The reverts commit has been added! @johnd0e

**Review by johnd0e — APPROVED** [review]

**Comment by Falke-Design** [comment]
@bozdoz please solve the conflicts

**Pull request merged**